### ISSUE TITLE
ZoneCaptureCoalition:DrawZone incorrect display of drawn zones

### ISSUE THREAD

**Issue by Fillinfront** [FlightControl-Master/MOOSE#2052]
Sorry for the English, I'm using a translator

When I start a mission, zones are drawn on the map and everything is very good, but after many events occur in this zone, for example onEntryGuarded, onEnterEmpty, onEnterAtacked and so on, all the drawn zones from drawZone seem to overlap each other, thereby 10 such events it is impossible to see what vehicles, missiles or aircraft are in this zone.

To put it simply, the zone ceases to be transparent and seems to be filled with a solid opaque fill after a certain number of events. Below I attach the lua file that I use in the mission

[Uploading zone.txt…]()

**Comment by Applevangelist** [comment]
Looking at the code, the zones are undrawen on each change and then redrawn. I cannot see the code that you tried to upload - maybe something in there

**Comment by Fillinfront** [comment]
[zone3.txt](https://github.com/FlightControl-Master/MOOSE/files/13542145/zone3.txt)

I am attaching the code, as well as screenshots to demonstrate the problem, the first one shows the zone after 10 events in it, the second one shows the normal display at the beginning of the mission

![image](https://github.com/FlightControl-Master/MOOSE/assets/84450360/f9df48fe-f63f-45e1-b24c-a344c8f1e981)
![image](https://github.com/FlightControl-Master/MOOSE/assets/84450360/bb570a00-84e4-486b-ba86-23c7a031daad)

**Comment by Applevangelist** [comment]
Ok I see. The demo does not undraw the zone on a change as this was meant as a demo. That would need to be added
